examples/medplum-scheduling-demo/src/components/CreateEncounter.tsx

```diff
@@ -0,0 +1,115 @@
+import { Modal } from '@mantine/core';
+import { showNotification } from '@mantine/notifications';
+import { createReference, getQuestionnaireAnswers, normalizeErrorString } from '@medplum/core';
+import {
+  Appointment,
+  Coding,
+  Encounter,
+  Patient,
+  Practitioner,
+  Questionnaire,
+  QuestionnaireResponse,
+} from '@medplum/fhirtypes';
+import { QuestionnaireForm, useMedplum, useMedplumProfile } from '@medplum/react';
+import { IconCircleCheck, IconCircleOff } from '@tabler/icons-react';
+import { useNavigate } from 'react-router-dom';
+
+interface CreateEncounterProps {
+  appointment: Appointment;
+  patient: Patient;
+  readonly opened: boolean;
+  readonly handlers: {
+    readonly open: () => void;
+    readonly close: () => void;
+    readonly toggle: () => void;
+  };
+}
+
+export function CreateEncounter(props: CreateEncounterProps): JSX.Element {
+  const { appointment, patient, opened, handlers } = props;
+  const medplum = useMedplum();
+  const profile = useMedplumProfile() as Practitioner;
+  const navigate = useNavigate();
+
+  async function handleQuestionnaireSubmit(formData: QuestionnaireResponse): Promise<void> {
+    const answers = getQuestionnaireAnswers(formData);
+    const startDateTime = answers['start-date'].valueDateTime as string;
+    const endDateTime = answers['end-date'].valueDateTime as string;
+    const encounterClass = answers['class'].valueCoding as Coding;
+
+    const encounterData: Encounter = {
+      resourceType: 'Encounter',
+      status: 'finished',
+      appointment: [createReference(appointment)],
+      class: encounterClass,
+      serviceType: appointment.serviceType?.[0],
+      period: {
+        start: startDateTime,
+        end: endDateTime,
+      },
+      subject: createReference(patient),
+      participant: [
+        {
+          // Uses the logged user as the attender
+          individual: createReference(profile),
+          type: [{ coding: [{ system: 'http://terminology.hl7.org/CodeSystem/v3-ParticipationType', code: 'ATND' }] }],
```

Comment:
I wasn't sure if I should use ATND or PPRF

Reply:
ATND is fine